src/lightning/fabric/strategies/xla.py

```diff
@@ -72,16 +70,34 @@ def root_device(self) -> torch.device:
     def num_processes(self) -> int:
         return len(self.parallel_devices) if self.parallel_devices is not None else 0
 
-    @property
-    def checkpoint_io(self) -> CheckpointIO:
-        if self._checkpoint_io is None:
-            self._checkpoint_io = XLACheckpointIO()
-        return self._checkpoint_io
+    @property  # type: ignore[override]
+    def checkpoint_io(self) -> XLACheckpointIO:
```

Comment:
I think the return type could remain CheckpointIO, and returning a subtype would be ok. Then you wouldn't need to ignore the override error

Reply:
I think it's simpler to override the definition so that we don't need to assert later in the code when we use these classes. XLACheckpointIO doesn't have any special APIs or attributes but XLAPrecision does. The CheckpointIO typing change here follows what is being done for precision.